src/audio/codec_adapter/codec_adapter.c

```diff
@@ -308,7 +308,8 @@ int codec_adapter_params(struct comp_dev *dev,
 
 	ret = memcpy_s(&mod->stream_params, sizeof(struct sof_ipc_stream_params),
 		       params, sizeof(struct sof_ipc_stream_params));
-	assert(!ret);
+	if (ret < 0)
+		return ret;
```

Comment:
this doesn't change anything, right? The only way for `memcpy_s()` above to fail would be if `params` were overlapping with `&mod->stream_params` which would probably be a bad bug deserving a panic anyway...

Reply:
@ranj063 suggested to remote the assert.